File: frontend/src/services/notificationService.ts
import { apiClient } from '@/config/apiClient';
import type { ApiResponse, PageResponse, TableParams } from '@/types';

export interface NotificationResponse {
  id: number;
  title: string;
  content: string;
  type: string;
  isRead: boolean;
  createdAt: string;
}

export const notificationService = {
  getMyNotifications: (params?: TableParams): Promise<ApiResponse<PageResponse<NotificationResponse>>> => {
    return apiClient.get('/notifications', { params });
  },

  getUnreadCount: (): Promise<ApiResponse<number>> => {
    return apiClient.get('/notifications/unread-count');
  },

  markAsRead: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.put(`/notifications/${id}/read`);
  },

  markAllAsRead: (): Promise<ApiResponse<void>> => {
    return apiClient.put('/notifications/read-all');
  },
};

File: backend/src/main/java/com/bubbleflow/service/NotificationService.java
package com.bubbleflow.service;

import com.bubbleflow.dto.notification.response.NotificationResponse;
import com.bubbleflow.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    PageResponse<NotificationResponse> getMyNotifications(Pageable pageable);
    long getUnreadCount();
    void markAsRead(Long id);
    void markAllAsRead();
    void createNotification(String title, String content, String type, Long userId);
}
